starlette/templating.py

```diff
@@ -56,9 +56,9 @@ class Jinja2Templates:
 
     def __init__(self, directory: str) -> None:
         assert jinja2 is not None, "jinja2 must be installed to use Jinja2Templates"
-        self.env = self.get_env(directory)
+        self.env = self._get_env(directory)
 
-    def get_env(self, directory: str) -> "jinja2.Environment":
+    def _get_env(self, directory: str) -> "jinja2.Environment":
```

Comment:
@JayH5 I've been thinking about this kind of things:

```suggestion
    def get_env(self, directory: str) -> "jinja2.Environment":
        warnings.warn(
            "'get_env' is deprecated. Use '_get_env' instead.", 
            DeprecationWarning
        )
        return self._get_env(directory)

    def _get_env(self, directory: str) -> "jinja2.Environment":
```
We did it for GraphQL, but we are in a pre-stable version, so in theory we can be aggressive and remove without adding warnings :thinking:

Reply:
I think the whole idea is to shy away from using `get_env` direcrly. We can change the warning to encourage using `templates.env` instead of `templates.get_env()`

Reply:
We definitely wouldn't want to suggest using a private method. We could say `[...] Use the env attribute instead` but even that is not equivalent to `get_env`.

If we wanted to be really careful we could rename this method to `create_env` and then call it from `get_env` with a warning like "this is deprecated, you probably want the env attribute, but if not use create_env". But, IMO, people should basically never want what `get_env` currently does and it's only confusing. I agree it should be removed.

Reply:
> We definitely wouldn't want to suggest using a private method.

Yeah, I agree! I've copied the warning from another snippet. 😛 

But my intention here was to bring the discussion if we should keep adding warnings in pre-1.0.